labgrid/driver/power/netio_json.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Add HTTP-JSON-API functions for labgrid-netio power driver.
+Set model in YML-CFG to 'netio_json'.
+"""
+__author__ = "Eugen Wiens, Christian Happ, Raffael Krakau"
+__copyright__ = "Copyright 2023, JUMO GmbH & Co. KG"
+__email__ = "Eugen.Wiens@jumo.net, Christian.Happ@jumo.net"
+__date__ = "2023-07-05"
+__status__ = "Production"
+
+import json
+
+import requests
+
+PORT = 80
+
+
+class __NetioControl:
+    """
+    Netio-Class to switch sockets via HTTP-JSON-API
+    """
+    __host: str
+    __port: int
+    __username: str
+    __password: str
+
+    def __init__(self, host, port=80, username="", password=""):
+        self.__host = host
+        self.__port = port
+        self.__username = username
+        self.__password = password
+
+    def set_state(self, socketID, action):
+        """
+        Set power state by socket port number (e.g. 1 - 8) and an action ('1'-'4').
+
+        - actions:
+            - 0: Turn OFF,
+            - 1: Turn ON,
+            - 2: Short OFF delay (restart),
+            - 3: Short ON delay,
+            - 4: Toggle (invert the state)
+        """
+        state = None
+        response = requests.post(self.generate_request_url(), json=self.get_request_json_data(socketID, action))
+
+        if response.ok:
+            responseDict = json.loads(response.text)
+            outputStates = responseDict['Outputs']
+
+            for outputState in outputStates:
+                if outputState['ID'] == socketID:
+                    state = outputState
+                    break
+        else:
+            raise Exception(f"Cannot SET the power state for socket number: {socketID} for action: {action}."
+                            f" Error code: {response.text}")
+
+        return state
+
+    def get_state(self, socketID):
+        """
+        Get current state of a given socket number as json.
+        """
+        state = None
+        response = requests.get(self.generate_request_url())
+
+        if response.ok:
+            responseDict = json.loads(response.text)
+
+            for outputState in responseDict['Outputs']:
+                if outputState['ID'] == socketID:
+                    state = outputState
```

Comment:
This should be
```state = outputState['State']```
Otherwise it always shows as on

Reply:
thank you